### PR TITLE
Report exceptional feed failures as FeedException.

### PR DESCRIPTION
**Pull request by bjorncs**
Change signature of ResultCallback to use FeedException.
Rename 'JsonParseException' => 'OperationParseException'.
Add exception for response/result parse failure.

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java [hunk 1]
```diff
@@ -63,12 +63,12 @@ public void dispatch(HttpRequest wrapped, CompletableFuture<HttpResponse> vessel
             endpoint.client.execute(request,
                                     new FutureCallback<SimpleHttpResponse>() {
                                         @Override public void completed(SimpleHttpResponse response) { vessel.complete(new ApacheHttpResponse(response)); }
-                                        @Override public void failed(Exception ex) { vessel.completeExceptionally(ex); }
+                                        @Override public void failed(Exception ex) { vessel.completeExceptionally(new FeedException(ex)); }
```
Comment (jonmv) [inline vespa-feed-client/src/main/java/ai/vespa/feed/client/ApacheCluster.java:66]: We can do this conversion higher up, where we have the document ID.

**Review by jonmv — COMMENTED** [review]
One comment, otherwise LGTM :)

**Comment by bjorncs** [comment]
@jonmv PTAL

**Comment by bjorncs** [comment]
@jonmv PTAL

**Review by jonmv — APPROVED** [review]
Yes!

**Pull request merged**